fetchmap/index.ts

```diff
@@ -0,0 +1,265 @@
+  layers: LayerDescriptor[];
+  popupSettings: FetchMapResult['popupSettings'] | null;
+  initialViewState?: Deck['props']['initialViewState'];
+}
+
+let currentMapData: AggregatedMapData | null = null; // Stores aggregated data from selected maps
+
+// Base options for fetchMap, cartoMapId will be added per call
+const baseFetchMapOptions = {
+  apiBaseUrl
+  // accessToken: import.meta.env.VITE_CARTO_ACCESS_TOKEN // if you want to use a private (non-public) map
+};
+
+// HTML Elements
+const mapNameEl = document.querySelector<HTMLDListElement>('#mapName');
+const layerCountEl = document.querySelector<HTMLDListElement>('#layerCount');
+const mapSelectorContainer = document.getElementById('map-selector-container');
+const loadMapsButton = document.getElementById('load-maps-button');
+
+// Initial Deck.gl view state - this will be updated by fetchMap data
+const INITIAL_VIEW_STATE = {
+  latitude: 0,
+  longitude: 0,
+  zoom: 1,
+  bearing: 0,
+  pitch: 0
+};
+
+const deck = new Deck({
+  canvas: 'deck-canvas',
+  initialViewState: INITIAL_VIEW_STATE,
+  controller: true,
+  onViewStateChange: ({viewState}) => {
+    const {longitude, latitude, ...rest} = viewState;
+    map.jumpTo({center: [longitude, latitude], ...rest});
+  },
+  getTooltip: ({object, layer}) => {
+    if (!layer) return null;
+    return buildTooltip(object, layer, currentMapData);
+  }
+});
+
+// Add basemap
+const map = new maplibregl.Map({
+  container: 'map',
+  style: BASEMAP.VOYAGER, // Using Voyager as a default
+  interactive: false
+});
+
+async function initialize(selectedMapIds: string[]) {
+  try {
+    // Clear previous map state
+    const existingLegend = document.querySelector('.legend-wrapper');
+    if (existingLegend) {
+      existingLegend.remove();
+    }
+    deck.setProps({layers: []});
+    currentMapData = null;
+
+    if (selectedMapIds.length === 0) {
+      if (mapNameEl) mapNameEl.innerHTML = 'No Map Selected';
+      if (layerCountEl) layerCountEl.innerHTML = '0';
+      map.jumpTo({
+        center: [INITIAL_VIEW_STATE.longitude, INITIAL_VIEW_STATE.latitude],
+        zoom: INITIAL_VIEW_STATE.zoom,
+        bearing: INITIAL_VIEW_STATE.bearing,
+        pitch: INITIAL_VIEW_STATE.pitch
+      });
+      deck.setProps({initialViewState: INITIAL_VIEW_STATE});
+      return;
+    }
+
+    // Fetch map data for each selected map
+    const allFetchedMapData: FetchMapResult[] = [];
+    for (const mapId of selectedMapIds) {
+      try {
+        const fetchOptions = {...baseFetchMapOptions, cartoMapId: mapId};
+        const mapData = await fetchMap(fetchOptions);
+        allFetchedMapData.push(mapData);
+      } catch (error) {
+        // Optionally, show a partial error or skip this map
+      }
+    }
+
+    // If no maps were fetched, show an error
+    if (allFetchedMapData.length === 0) {
+      if (mapNameEl) mapNameEl.innerHTML = 'Error loading maps';
+      if (layerCountEl) layerCountEl.innerHTML = 'Error';
+      return;
+    }
```

Comment:
We're already inside of a larger try/catch block here, not sure the inner try/catch  in the loop is needed? Hiding the failed requests is probably not what we'd recommend I think. Lines 96–112 could be simplified to:

```javascript
const allFetchedMapData = await Promise.all(
  selectedMapIds.map((cartoMapId) => fetchMap({...baseFetchMapOptions, cartoMapId}))
);
```

... with the extra benefit that the requests run in parallel rather than sequentially, so loading should be faster.